ocpp/v201/enums.py

```diff
@@ -0,0 +1,1193 @@
+    FirmwareStatusNotification = "FirmwareStatusNotification"
+    Get15118EVCertificate = "Get15118EVCertificate"
+    GetBaseReport = "GetBaseReport"
+    GetCertificateStatus = "GetCertificateStatus"
+    GetChargingProfiles = "GetChargingProfiles"
+    GetCompositeSchedule = "GetCompositeSchedule"
+    GetDisplayMessages = "GetDisplayMessages"
+    GetInstalledCertificateIds = "GetInstalledCertificateIds"
+    GetLocalListVersion = "GetLocalListVersion"
+    GetLog = "GetLog"
+    GetMonitoringReport = "GetMonitoringReport"
+    GetReport = "GetReport"
+    GetTransactionStatus = "GetTransactionStatus"
+    GetVariables = "GetVariables"
+    Heartbeat = "Heartbeat"
+    InstallCertificate = "InstallCertificate"
+    LogStatusNotification = "LogStatusNotification"
+    MeterValues = "MeterValues"
+    NotifyChargingLimit = "NotifyChargingLimit"
+    NotifyCustomerInformation = "NotifyCustomerInformation"
+    NotifyDisplayMessages = "NotifyDisplayMessages"
+    NotifyEVChargingNeeds = "NotifyEVChargingNeeds"
+    NotifyEVChargingSchedule = "NotifyEVChargingSchedule"
+    NotifyEvent = "NotifyEvent"
+    NotifyMonitoringReport = "NotifyMonitoringReport"
+    NotifyReport = "NotifyReport"
+    PublishFirmware = "PublishFirmware"
+    PublishFirmwareStatusNotification = "PublishFirmwareStatusNotification"
+    ReportChargingProfiles = "ReportChargingProfiles"
+    RequestStartTransaction = "RequestStartTransaction"
+    RequestStopTransaction = "RequestStopTransaction"
+    ReservationStatusUpdate = "ReservationStatusUpdate"
+    ReserveNow = "ReserveNow"
+    Reset = "Reset"
+    SecurityEventNotification = "SecurityEventNotification"
+    SendLocalList = "SendLocalList"
+    SetChargingProfile = "SetChargingProfile"
+    SetDisplayMessage = "SetDisplayMessage"
+    SetMonitoringBase = "SetMonitorBase"
+    SetMonitoringLevel = "SetMonitoringLevel"
+    SetNetworkProfile = "SetNetworkProfile"
+    SetVariableMonitoring = "SetVariableMonitoring"
+    SetVariables = "SetVariables"
+    SignCertificate = "SignCertificate"
+    StatusNotification = "StatusNotification"
+    TransactionEvent = "TransactionEvent"
+    TriggerMessage = "TriggerMessage"
+    UnlockConnector = "UnlockConnector"
+    UnpublishFirmware = "UnpublishFirmware"
+    UpdateFirmware = "UpdateFirmware"
+
+# Enums
+
+
+class APNAuthenticationType:
+    """
+    APNAuthenticationEnumType is used by:
+    setNetworkProfile:SetNetworkProfileRequest.APNType
+    """
+    chap = "CHAP"
+    none = "NONE"
+    pap = "PAP"
+    auto = "AUTO"
+
+
+class AttributeType:
+    """
+    AttributeEnumType is used by: Common:VariableAttributeType ,
+    getVariables:GetVariablesRequest.GetVariableDataType ,
+    getVariables:GetVariablesResponse.GetVariableResultType ,
+    setVariables:SetVariablesRequest.SetVariableDataType ,
+    setVariables:SetVariablesResponse.SetVariableResultType
+    """
+    actual = "Actual"
+    target = "Target"
+    minSet = "MinSet"
+    maxSet = "MaxSet"
```

Comment:
Can you use `snake_case` for attributes?

Reply:
In snake_case, what would be correct in this situation:
```python
re.sub(r'(?<!^)(?=[A-Z])', '_', "ALBama").lower()
> 'a_l_bama'

or 

re.sub('([a-z0-9])([A-Z])', r'\1_\2', "ALABama").lower()
> 'alabama'
```

Reply:
`snake_case` separates different words or terms with `_`. The problem with `ALABama` is that it is a single word with weird capitalization. The correct outcome is `alabama`.